internal/graph/cached_resolver_test.go

```diff
@@ -0,0 +1,192 @@
+package graph
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResolveCheck(t *testing.T) {
+	ctx := context.Background()
+
+	req := &ResolveCheckRequest{
+		StoreID:              "12",
+		AuthorizationModelID: "33",
+		TupleKey: &openfgav1.TupleKey{
+			Object:   "document:abc",
+			Relation: "reader",
+			User:     "user:XYZ",
+		},
+	}
+	result := &ResolveCheckResponse{Allowed: true}
+
+	// if the tuple is different, it should result in fetching from cache
+	tests := []struct {
+		_name   string
+		req     *ResolveCheckRequest
+		prepare func(mock *MockCheckResolver, request *ResolveCheckRequest)
+	}{
+		{
+			// same signature means data will be taken from cache
+			_name: "same_signature",
+			req: &ResolveCheckRequest{
+				StoreID:              "12",
+				AuthorizationModelID: "33",
+				TupleKey: &openfgav1.TupleKey{
+					Object:   "document:abc",
+					Relation: "reader",
+					User:     "user:XYZ",
+				},
+			},
+			prepare: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				// there should be no call
+			},
+		},
+		{
+			// different store means data is not from cache
+			_name: "different_store",
```

Comment:
```suggestion
			_name: "request_for_different_store_does_not_return_results_from_cache",
```

same suggestion for the remaining tests